Iis2dlpc move drdy_int and tap configuration into device tree

avisconti opened the pull request:
(Depend on #30317).
Move some informations from KConfig to Device Tree..
  - drdy_int.
  - pulse (renamed to tap).
.
As agreed in #30317 the name has been changed from 'pulse' to 'tap' and the tap-ths property value is used to enable/disable the feature for each axis (> 0 enables it). The dts binding is describing each property with particular care for the default value..

Tested on x_nucleo_iks02a1 shield.

avisconti committed:
drivers/sensor: iis2dlpc: Move drdy_int info into DT

The IIS2DLPC drdy interrupt can be routed to either INT1 or
INT2 pin. Currently the selection is done by Kconfig configuration.
This commit is instead moving it into Device Tree as 'drdy-int'.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

pabigot commented on dts/bindings/sensor/st,iis2dlpc-common.yaml:
```diff
@@ -11,6 +11,17 @@ properties:
         The property value should ensure the flags properly describe
         the signal that is presented to the driver.
 
+    drdy-pin:
```
How about `drdy-int`?  `foo-pin` is currently used in Nordic boards to identify a GPIO ordinal; I could also see it being used in other contexts as a pad (pin) identifier.

Alternatively, since there are only two choices:  `drdy-int2` as a boolean property; if not selected then it's `INT1`.  This would eliminate the need to add build-time value validation.  (Though so would making this an int enum accepting only 1 and 2.

avisconti replied:
> How about `drdy-int`? `foo-pin` is currently used in Nordic boards to identify a GPIO ordinal; I could also see it being used in other contexts as a pad (pin) identifier.
> 
drdy-int seems a better name, thanks!

> Alternatively, since there are only two choices: `drdy-int2` as a boolean property; if not selected then it's `INT1`. This would eliminate the need to add build-time value validation. (Though so would making this an int enum accepting only 1 and 2.

To be honest I don't like the idea of having a boolean like that. It seems to me more clear an attribute that specify either 1 or 2. And yes, I will add the two possible values using the 'enum' declaration.

Does it make sense to you?

pabigot replied:
That'd be fine; the main thing is to make sure people can't configure it to an invalid state.

avisconti replied:
When specifying enum a build-time checking, right? At least I tried with 'bad' values and didn't build up.

pabigot commented on dts/bindings/sensor/st,iis2dlpc-common.yaml:
```diff
@@ -44,3 +44,76 @@ properties:
+        - 0 # Only Single Tap
+        - 1 # Single and Double Tap
+
+    tap-thsx:
```
Based on other recent discussions, how about making this type `array` and documenting that it has three values corresponding to the X, Y, and Z threshold settings, and noting that a value of 0 means the axis is disabled.

Same for any other spatial properties (though I don't see any right now).

avisconti replied:
OK, I'm acking also this one. Seems easy and it simplifies the code.

pabigot commented on dts/bindings/sensor/st,iis2dlpc-common.yaml:
```diff
@@ -44,3 +44,76 @@ properties:
+        detection to be recognized as a tap event. Where 0 equals 4*1/ODR and
+        1LSB = 8*1/ODR.
+
+    tap-ltcy:
```
maybe spell out `latency`?

avisconti replied:
Sounds good. Since my native language (italian) is so rich of vowels I have developed a special art in cutting them out and using only consonant to have shorter names. But here seems too much in fact. :)

pabigot reviewed (changes requested):
Several suggestions.  The strongest one is to use an array for the spacial threshold configuration, but I could be convinced to back off that.

avisconti commented:
Added the DNM because this PR depends to #30317.

pabigot commented on drivers/sensor/iis2dlpc/iis2dlpc.c:
```diff
@@ -380,21 +393,15 @@ const struct iis2dlpc_device_config iis2dlpc_cfg = {
 
+#ifdef CONFIG_IIS2DLPC_TAP
+	.tap_mode = DT_INST_PROP(0, tap_mode),
+	.tap_ths = DT_INST_PROP(0, tap_ths),
```
This could be `tap_threshold`.

avisconti replied:
sure

avisconti replied:
BTW, this is the name in the datasheet, but there is no reason to keep it same in driver

pabigot commented on dts/bindings/sensor/st,iis2dlpc-common.yaml:
```diff
@@ -47,3 +47,65 @@ properties:
+        - 0 # Only Single Tap
+        - 1 # Single and Double Tap
+
+    tap-ths:
```
`tap-threshold` ?

avisconti replied:
Sure. (Curious about the slang term...)

pabigot reviewed (changes requested):
Looks pretty good thanks.  I think it's worth spelling out threshold, because it'd match the other devicetree properties and because the natural pronunciation of `tap_ths` suggests an inappropriate English slang term.

avisconti commented:
Rebased it soon after #30317 has been merged on master. And removed DNM.

mbolivar-nordic reviewed (approved):
Nice cleanup!

avisconti commented:
@mhelm Do you happen to have some tme to take a look at this one today?

MaureenHelm commented on drivers/sensor/iis2dlpc/iis2dlpc_trigger.c:
```diff
@@ -29,46 +29,47 @@ static int iis2dlpc_enable_int(const struct device *dev,
 	iis2dlpc_reg_t int_route;
 
-	if (cfg->drdy_int == 1U) {
+	switch (type) {
+	case SENSOR_TRIG_DATA_READY:
+		if (cfg->drdy_int == 1) {
+			/* set interrupt for pin INT1 */
+			iis2dlpc_pin_int1_route_get(iis2dlpc->ctx,
+					&int_route.ctrl4_int1_pad_ctrl);
+			int_route.ctrl4_int1_pad_ctrl.int1_drdy = enable;
+
+			return iis2dlpc_pin_int1_route_set(iis2dlpc->ctx,
+					&int_route.ctrl4_int1_pad_ctrl);
+		} else {
+			/* set interrupt for pin INT2 */
+			iis2dlpc_pin_int2_route_get(iis2dlpc->ctx,
+					&int_route.ctrl5_int2_pad_ctrl);
+			int_route.ctrl5_int2_pad_ctrl.int2_drdy = enable;
+
+			return iis2dlpc_pin_int2_route_set(iis2dlpc->ctx,
+					&int_route.ctrl5_int2_pad_ctrl);
+		}
+		break;
```
It looks like you're mixing drdy_int changes with tap/pulse changes? Can you move this section to the commit `drivers/sensor: iis2dlpc: Move drdy_int info into DT`

avisconti replied:
> It looks like you're mixing drdy_int changes with tap/pulse changes? Can you move this section to the commit `drivers/sensor: iis2dlpc: Move drdy_int info into DT`

mmmmh, actually it is neither belonging to the first one nor to the second one. It is just a different way to handle the switch of trigger types. But you are right in saying that does no9t belong to the TAP changes. So, if you agree, I can move this change to an intermediate commit.

MaureenHelm replied:
Yes, please

avisconti replied:
OK, splitted the work into 3 commits. The intermediate one is carrying the "switch change", which is done just for a better readability.

MaureenHelm reviewed (commented):
Nice code cleanup

avisconti committed:
drivers/sensor: iis2dlpc: Simplify the switch on trigger type

Simplify the switch case on trigger types (SENSOR_TRIG_DRDY and
SENSOR_TRIG_TAP) inside iis2dlpc_enable_int().

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: iis2dlpc: Move trigger pulse Kconfig property into DT

Move iis2dlpc trigger pulse configurations from Kconfigs to Device Tree.
Moreover the dts properties have been renamed as 'tap', which sounds a
better name to immediately catch the feature behind it. Since tap
threshold cannot be zero, this value (which is the default in dts
binding) is used to enable/disable the device feature per each axis.
The event can be generated on INT1 only.

Signed-off-by: Armando Visconti <armando.visconti@st.com>
